### PR TITLE
Use BinaryData whenever the type is AnyObject for mgmt plane

### PR DESCRIPTION
**Pull request by m-nash**
# Description
For now this will be only for mgmt plane but in the future we will want this for some dataplane APIs.  In order to do that in a backwards compatible way we will need to add a new flag so each SDK can opt in and potentially have the configuration at a model level.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/OperationSourceWriter.cs [hunk 1]
```diff
@@ -135,13 +135,27 @@ public void WriteCreateResult(string responseVariable)
 
             using (_writer.Scope($"{_opSource.ReturnType} {_opSource.Interface}.CreateResult({typeof(Response)} {responseVariable:D}, {typeof(CancellationToken)} cancellationToken)"))
             {
-                _writer.WriteDeserializationForMethods(_opSource.ResponseSerialization, false, valueCallback, responseVariable);
+                //if (_opSource.ReturnType.Equals(typeof(BinaryData)))
+                //{
+                //    _writer.Line($"return {typeof(BinaryData)}.FromStream(response.ContentStream);");
+                //}
+                //else
+                //{
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/Mgmt/Generation/OperationSourceWriter.cs:143]: If this code is not needed, can we remove it?

File: test/TestProjects/MgmtScopeResource/Generated/Models/ParameterValuesValue.cs [hunk 2]
```diff
@@ -17,12 +19,12 @@ public ParameterValuesValue()
...
         /// <param name="value"> The value of the parameter. </param>
-        internal ParameterValuesValue(object value)
+        internal ParameterValuesValue(BinaryData value)
         {
             Value = value;
         }
 
         /// <summary> The value of the parameter. </summary>
-        public object Value { get; set; }
+        public BinaryData Value { get; set; }
```
Comment (ShivangiReja) [inline test/TestProjects/MgmtScopeResource/Generated/Models/ParameterValuesValue.cs:28]: Can we add tests to verify that it is generated as BinaryData?

**Review by ShivangiReja — APPROVED** [review]
LGTM

**Review by annelo-msft — APPROVED** [review]
LGTM - I see no changes to data plane libraries, and we can work on scoping for data plane in the future.

**Pull request merged**